llvm-build/libllvm.ps1

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+. (Join-Path $PSScriptRoot .. build "set-env.ps1")
+. (Join-Path $PSScriptRoot "utils.ps1")
+
+Assert (Test-Path env:\PKG_NAME) "PKG_NAME was not defined"
+Assert (Test-Path env:\LINUX_PKG_NAME) "LINUX_PKG_NAME was not defined"
+Assert (Test-Path env:\WINDOWS_PKG_NAME) "WINDOWS_PKG_NAME was not defined"
+Assert (Test-Path env:\DARWIN_PKG_NAME) "DARWIN_PKG_NAME was not defined"
+Assert (Test-Path env:\BLOBS_OUTDIR) "BLOBS_OUTDIR was not defined"
+Assert (Test-Path env:\BUILD_ARTIFACTSTAGINGDIRECTORY) "BUILD_ARTIFACTSTAGINGDIRECTORY was not defined"
+
+$llvmArchiveFiles = @(
+    "$($env:LINUX_PKG_NAME)/llvm-build/llvm-project/build/$($env:LINUX_PKG_NAME).tar.gz",
+    "$($env:WINDOWS_PKG_NAME)/llvm-build/llvm-project/build/$($env:WINDOWS_PKG_NAME).zip",
+    "$($env:DARWIN_PKG_NAME)/llvm-build/llvm-project/build/$($env:DARWIN_PKG_NAME).tar.gz"
+)
+
+for ($i = 0; $i -lt $llvmArchiveFiles.Count; $i++) {
+    $file = $llvmArchiveFiles[$i]
+    Write-Vso "Raw: $file"
+    $path = Join-Path (Get-Location) '..' $file
+    Write-Vso "Joined: $path"
+    $resolvedPath = Resolve-Path $path
+    Write-Vso "Resolved: $resolvedPath"
+    $llvmArchiveFiles[$i] = $resolvedPath
+}
+
+$artifactsDir = $env:BLOBS_OUTDIR
+if (!(Test-Path $artifactsDir)) {
+    mkdir $artifactsDir | Out-Null
+}
+$artifactsDir = Resolve-Path "$artifactsDir"
+
+foreach ($file in $llvmArchiveFiles) {
+    $file = (Get-Item $file)
+    $extension = $file.Extension
+    if (".zip" -eq $extension) {
+        exec { 7z x $file }
+    }
+    elseif (".gz" -eq $extension) {
+        $tarGz = $file
+        Write-Vso "Processing $tarGz"
+        exec { 7z x $tarGz }
+        $tar = $tarGz.Name.TrimEnd($extension)
+        Write-Vso "Processing $tar"
+        exec { 7z x -aoa -ttar $tar }
+    }
+}
+
+# Windows: bin/LLVM-C.dll
+# Linux: lib/libLLVM-11.so
+# Darwin: lib/libLLVM.dylib
+
+$linuxPath = Resolve-Path (Join-Path  '.' "$($env:LINUX_PKG_NAME)" "lib" "libLLVM-[0-9][0-9].so")
+Write-Vso "Test Linux Path: $linuxPath"
+Test-Path $linuxPath
+
+$windowsPath = Resolve-Path (Join-Path  '.' "$($env:WINDOWS_PKG_NAME)" "bin" "LLVM-C.dll")
+Write-Vso "Test Windows Path: $windowsPath"
+Test-Path $windowsPath
+
+$darwinPath = Resolve-Path (Join-Path  '.' "$($env:DARWIN_PKG_NAME)" "lib" "libLLVM.dylib")
+Write-Vso "Test Darwin Path: $darwinPath"
+Test-Path $darwinPath
```

Comment:
(I'm not very familiar with pwsh)
What does change if any of these `Test-Path` commands returns `$False`? (What do these lines 56 - 66 do?)

Reply:
This won't fail the script execution, but it does output `true` or `false` into the build log. See for example here:
https://dev.azure.com/ms-quantum-public/Microsoft%20Quantum%20(public)/_build/results?buildId=36741&view=logs&j=b1b4ecf1-bb33-53e2-fb11-822bd4878653&t=0625e23a-d1dd-514b-4717-f10b81a8af4e&l=119